gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/mapper/UserMapper.java

```diff
@@ -77,7 +79,7 @@ public static User convert(io.gravitee.am.model.User user, String baseUrl, boole
         scimUser.setPhotos(toScimAttributes(user.getPhotos()));
         scimUser.setAddresses(toScimAddresses(user.getAddresses()));
         scimUser.setEntitlements(user.getEntitlements());
-        scimUser.setRoles(user.getRoles());
+        scimUser.setRoles(getRoleStream(user).flatMap(List::stream).sorted().distinct().collect(toList()));
```

Comment:
With `getRoleStream` doesn't flatten the two list? 
I would prefer a `getAllRoles` method that includes the processing flow `flatMap(List::stream).sorted().distinct().collect(toList()))` as this method is only used here.